rusk-wallet/src/lib/cache.rs

```diff
@@ -0,0 +1,96 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+
+use dusk_bytes::{DeserializableSlice, Serializable};
+use phoenix_core::Note;
+use rusqlite::{params, Connection};
+
+pub(crate) fn open(data_dir: &PathBuf) -> Connection {
+    let db_path = Path::new(data_dir).join("cache.db");
+
+    let cache = Connection::open(db_path).unwrap();
+    //create table if not exists TableName (col1 typ1, ..., colN typN)
+    cache
+        .execute(
+            "CREATE TABLE if not exists notes (note BLOB, spendkey BLOB)",
+            [],
+        )
+        .unwrap();
+    cache
+        .execute(
+            "CREATE TABLE if not exists cache (block BIGINT, spendkey BLOB)",
+            [],
+        )
+        .unwrap();
+    cache
+}
+
+pub(crate) fn last_blockheight(conn: &Connection, spendkey: &[u8]) -> u64 {
```

Comment:
Again, I'd suggest adding a small doc comment. You know what you're doing but it's not that obvious to someone looking at this for the first time.